scipy/signal/tests/test_signaltools.py

```diff
@@ -1997,34 +2197,62 @@ def _setup_rank3(self, dt):
 
         return a, b, y_r
 
-    def test_rank3_valid(self, dt):
-        a, b, y_r = self._setup_rank3(dt)
+    @skip_xp_backends(np_only=True, reasons=["order='F'"])
+    def test_rank3_valid(self, dt, xp):
+        dt = getattr(xp, dt) if isinstance(dt, str) else dt
+        a, b, y_r = self._setup_rank3(dt, xp)
         y = correlate(a, b, "valid")
-        assert_array_almost_equal(y, y_r[1:2, 2:4, 3:5])
-        assert_equal(y.dtype, dt)
+        _assert_almost_equal = self._get_assertion(dt)
+        _assert_almost_equal(y, y_r[1:2, 2:4, 3:5])
+        assert y.dtype == dt
 
         # See gh-5897
         y = correlate(b, a, "valid")
-        assert_array_almost_equal(y, y_r[1:2, 2:4, 3:5][::-1, ::-1, ::-1])
-        assert_equal(y.dtype, dt)
-
-    def test_rank3_same(self, dt):
-        a, b, y_r = self._setup_rank3(dt)
+        _assert_almost_equal = self._get_assertion(dt)
+        _assert_almost_equal(y, y_r[1:2, 2:4, 3:5][::-1, ::-1, ::-1])
+        assert y.dtype == dt
+
+    @skip_xp_backends(np_only=True, reasons=["order='F'"])
+    def test_rank3_same(self, dt, xp):
+        dt = getattr(xp, dt) if isinstance(dt, str) else dt
+        a, b, y_r = self._setup_rank3(dt, xp)
         y = correlate(a, b, "same")
-        assert_array_almost_equal(y, y_r[0:-1, 1:-1, 1:-2])
-        assert_equal(y.dtype, dt)
-
-    def test_rank3_all(self, dt):
-        a, b, y_r = self._setup_rank3(dt)
+        _assert_almost_equal = self._get_assertion(dt)
+        _assert_almost_equal(y, y_r[0:-1, 1:-1, 1:-2])
+        assert y.dtype == dt
+
+    @skip_xp_backends(np_only=True, reasons=["order='F'"])
+    def test_rank3_all(self, dt, xp):
+        dt = getattr(xp, dt) if isinstance(dt, str) else dt
+        a, b, y_r = self._setup_rank3(dt, xp)
         y = correlate(a, b)
-        assert_array_almost_equal(y, y_r)
-        assert_equal(y.dtype, dt)
+        _assert_almost_equal = self._get_assertion(dt)
+        _assert_almost_equal(y, y_r)
+        assert y.dtype == dt
+
+
+@pytest.mark.parametrize('dt', ["uint8", "int8", "uint16", "int16",
+                                "uint32", "int32", # np_ulong, np_long,   # FIXME: np 1.x & 2.x / backends?
+                                "uint64", "int64", # FIXME: is this correct? Was np.ulonglong, np.ulonglong,
```

Comment:
This looks correct. We should only be using the explicitly-sized dtype names. If we have `int32`, `uint32`, `int64`, `uint64`, that should cover everything.

Reply:
There was some story about `npy_{u}long` for numpy 1.x vs 2.x

Reply:
Sure, but it's an alias for either the 32-bit or the 64-bit sized dtype. There was a change on Windows with a default from `long` to `intp`, and maybe also some discussion around keeping `np.long` as the C-like dtype name. (and maybe something else - it was messy)

I don't think that is too relevant here. Incoming arrays are sized, so tests constructing with `np.ulong` & co isn't needed.
```python
>>> np.long
<class 'numpy.int64'>
>>> np.ulong
<class 'numpy.uint64'>
```